cookbooks/creating_a_behat_extension.rst

```diff
@@ -0,0 +1,270 @@
+              LogsInitializer.php   # This will handle context initialization
+          LogsContext.php
+      ServiceContainer/
+          LogsExtension.php
+
+The code for ``LogsInitializer.php``:
+
+.. code-block:: php  
+  <?php
+
+  namespace Behat\LogsExtension\Context\Initializer;
+
+  use Behat\LogsExtension\Context\LogsContext;
+  use Behat\Behat\Context\Context;
+  use Behat\Behat\Context\Initializer\ContextInitializer;
+
+  class LogsInitializer implements ContextInitializer
+  {
+      private string $filepath;
+      private bool $enable;
+  
+      public function __construct(string $filepath, bool $enable)
+      {
+          $this->filepath = $filepath;
+          $this->enable = $enable;
+      }
+  
+      public function initializeContext(Context $context)
+      {
+          if (!$context instanceof LogsContext) {
+              return;
+          }
+
+          $context->initializeConfig($this->enable, $this->filepath);
+      }
+  }
+
+We need to register the initializer definition within the Behat container through the ``LogsExtension``, ensuring it gets loaded:
+
+.. code-block:: php
+  <?php
+
+  // ...
+
+  use Symfony\Component\DependencyInjection\Definition;
+  use Behat\Behat\Context\ServiceContainer\ContextExtension;
+
+  class LogsExtension implements Extension
+  {
+      // ...
+  
+      public function load(ContainerBuilder $container, array $config)
+      {
+          $definition = new Definition(LogsInitializer::class, [
+              $config['filepath'],
+              $config['enable'],
+          ]);
+          $definition->addTag(ContextExtension::INITIALIZER_TAG);
+          $container->setDefinition('logs_extension.context_initializer', $definition);
+      }
+
+      // ...
+  }
+
+To complete the extension, we must add methods to ``LogsContext`` to receive the configuration values and use those in the hooks:
+
+.. code-block:: php
+
+  // ...
+  
+  class LogsContext implements Context
+  {
+      private bool $enable = false;
+      private string $filepath;
+  
+      public function initializeConfig(bool $enable, string $filepath)
+      {
+          $this->enable = $enable;
+          $this->filepath = $filepath;
+      }
+  
+      /** @BeforeScenario */
+      public function before(BeforeScenarioScope $scope)
+      {
+          if ($this->enable === false) {
+              return;
+          }
+
+          file_put_contents(
+              $this->filepath,
+              'START: ' . $scope->getScenario()->getTitle() . ' - ' . time() . PHP_EOL,
+              FILE_APPEND
+          );
+      }
+  
+      /** @AfterScenario */
+      public function after(AfterScenarioScope $scope)
+      {
+          if ($this->enable === false) {
+              return;
+          }
+  
+          file_put_contents(
+              $this->filepath,
+              'END: ' . $scope->getScenario()->getTitle() . ' - ' . time() . PHP_EOL,
+              FILE_APPEND
+          );
+      }
+  }
+
+Conclusion
```

Comment:
The only final thing would be a quick example of what you need in your project's `behat.yml` to use this extension - I think in this case you'll need to register both the extension and the LogsContext separately so it would be worth showing that?

Reply:
@acoulton done with f91f7c9